docs/rules/prefer-object-spread.md

```diff
@@ -0,0 +1,47 @@
+# Prefer use of an object spread over `Object.assign` (prefer-object-spread)
+
+When Object.assign is called using an object literal the first argument, this rule requires using the object spread syntax instead. This rule also warns on cases where an `Object.assign` call is made using a single argument that is an object literal, in this case, the `Object.assign` call is not needed.
```

Comment:
Small typo: "using an object literal the first argument" --> "using an object literal as the first argument".

Also, we generally prefer that the first section should only have background about object spread and `Object.assign`-- the Rule Details section is where the rule's purpose/method should be outlined.